alternator/executor.cc

```diff
@@ -3817,34 +3817,28 @@ static future<executor::request_return_type> do_query(service::storage_proxy& pr
     command->slice.options.set<query::partition_slice::option::allow_short_read>();
     auto query_options = std::make_unique<cql3::query_options>(cl, std::vector<cql3::raw_value>{});
-    query_options = std::make_unique<cql3::query_options>(std::move(query_options), std::move(paging_state));
+    query_options = std::make_unique<cql3::query_options>(std::move(query_options), std::move(old_paging_state));
     auto p = service::pager::query_pagers::pager(proxy, schema, selection, *query_state_ptr, *query_options, command, std::move(partition_ranges), nullptr);
 
-    return p->fetch_page(limit, gc_clock::now(), executor::default_timeout()).then(
-            [p = std::move(p), schema, cql_stats, partition_slice = std::move(partition_slice),
-             selection = std::move(selection), query_state_ptr = std::move(query_state_ptr),
-             attrs_to_get = std::move(attrs_to_get),
-             query_options = std::move(query_options),
-             filter = std::move(filter)] (std::unique_ptr<cql3::result_set> rs) mutable {
-        if (!p->is_exhausted()) {
-            rs->get_metadata().set_paging_state(p->state());
-        }
-        auto paging_state = rs->get_metadata().paging_state();
-        bool has_filter = filter;
-        auto [items, size] = describe_items(*selection, std::move(rs), std::move(attrs_to_get), std::move(filter));
-        if (paging_state) {
-            rjson::add(items, "LastEvaluatedKey", encode_paging_state(*schema, *paging_state));
-        }
-        if (has_filter){
-            cql_stats.filtered_rows_read_total += p->stats().rows_read_total;
-            // update our "filtered_row_matched_total" for all the rows matched, despited the filter
-            cql_stats.filtered_rows_matched_total += size;
-        }
-        if (is_big(items)) {
-            return make_ready_future<executor::request_return_type>(make_streamed(std::move(items)));
-        }
-        return make_ready_future<executor::request_return_type>(make_jsonable(std::move(items)));
-    });
+    std::unique_ptr<cql3::result_set> rs = co_await p->fetch_page(limit, gc_clock::now(), executor::default_timeout());
+    if (!p->is_exhausted()) {
+        rs->get_metadata().set_paging_state(p->state());
+    }
+    auto paging_state = rs->get_metadata().paging_state();
+    bool has_filter = filter;
+    auto [items, size] = describe_items(*selection, std::move(rs), std::move(attrs_to_get), std::move(filter));
+    if (paging_state) {
+        rjson::add(items, "LastEvaluatedKey", encode_paging_state(*schema, *paging_state));
+    }
+    if (has_filter){
+        cql_stats.filtered_rows_read_total += p->stats().rows_read_total;
+        // update our "filtered_row_matched_total" for all the rows matched, despited the filter
+        cql_stats.filtered_rows_matched_total += size;
+    }
+    if (is_big(items)) {
+        co_return executor::request_return_type(make_streamed(std::move(items)));
```

Comment:
nit: with co_return you should be able to skip  `executor::request_return_type(`